### PR TITLE
Add SPI prepare and SPI exec_plan for R clients

### PR DESCRIPTION
**Pull request by HaozhouWang**
1. R clients now support pg.spi.prepare and pg.spi.execp
   please follow the plr for usage
2. Update backend SPI process to support R SPI requests

### CODE REVIEW

**Comment by zhangh43** [comment]
+1

File: src/sqlhandler.c [hunk 1]
```diff
@@ -198,42 +198,53 @@ plcMessage *handle_sql_message(plcMsgSQL *msg, plcProcInfo *pinfo) {
...
+				} else {
+						lprintf(ERROR, "prepare type is bad, unexpected prepare sql type %d",
+						        msg->args[i].type.type);
 				}
-				parseTypeString(msg->args[i].type.typeName, &type_oid, &typemod);
 
-				plc_plan->argOids[i] = type_oid;
-				argTypes[i] = plc_get_datatype_from_oid(type_oid);
+				argTypes[i] = plc_get_datatype_from_oid(plc_plan->argOids[i]);
 			}
```
Comment (paul-guo-) [inline src/sqlhandler.c]: Why reformat the code below?

**Comment by stanlyxiang** [comment]
Before merge this PR, you should update submodule and commit it after this  [PR](https://github.com/greenplum-db/rclient/pull/13) is merged.

File: src/sqlhandler.c (hunk 1 above)
Comment (HaozhouWang) [reply src/sqlhandler.c]: Thanks, fixed

**Comment by HaozhouWang** [comment]
Thanks for review, updated submodule.

**Pull request merged**